src/main/java/com/microsoft/aad/msal4j/AbstractClientApplicationBase.java

```diff
@@ -603,6 +608,21 @@ public T autoDetectRegion(boolean val) {
             return self();
         }
 
+        /**
+         * Indicates that the library should attempt to fetch the instance discovery metadata from the specified Azure region.
+         *
+         * If the region is valid, token requests will be sent to the regional ESTS endpoint rather than the global endpoint.
+         * If region information could not be verified, the library will fall back to using the global endpoint, which is also
+         *  the default behavior if this value is not set.
+         *
+         * @param val boolean (default is false)
+         * @return instance of the Builder on which method was called
+         */
+        public T azureRegion(String val) {
+            azureRegion = val;
+            return self();
+        }
+
```

Comment:
what if  region is provided and autodetect is enabled ?

Reply:
I don't think there's an actual use case for using both, however the design doc doesn't say they should be mutually exclusive (i.e., we shouldn't throw an error or anything if the dev tries to use both). 

Autodetection happens anyway if a region is provided so the autodetect flag is pretty much irrelevant in that case, and the dev provided region takes precedence over the autodetected one.